lib/Access.php

```diff
@@ -156,19 +156,23 @@ private function getLDAP() {
 	/**
 	 * reads a given attribute for an LDAP record identified by a DN
 	 * @param string $dn the record in question
-	 * @param string $attr the attribute that shall be retrieved
+	 * @param string|array $attrs the attribute that shall be retrieved
 	 *        if empty, just check the record's existence
 	 * @param string $filter
 	 * @return array|false an array of values on success or an empty
 	 *          array if $attr is empty, false otherwise
 	 */
-	public function readAttribute($dn, $attr, $filter = 'objectClass=*') {
+	public function readAttributes($dn, $attrs, $filter = 'objectClass=*') {
```

Comment:
Maybe trying to cram this into this function was a bad idea. The code is already too complex. To read the addributes of a single dn we don't need paging and ranged requests.